### ISSUE TITLE
`no-member` error in annotations

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#6594]
### Bug description

First discovered in https://github.com/PyCQA/astroid/pull/1555

```py
# pylint: disable=missing-docstring,unused-argument
from __future__ import annotations

import ast

def func(node: ast.Match) -> None:  # no-member
    ...
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
test.py:6:15: E1101: Module 'ast' has no 'Match' member (no-member)
```

### Expected behavior

With `from __future__ import annotations`, the type annotation isn't evaluated at runtime and thus won't throw an error. Furthermore it's possible to use classes that way which are only defined in stub files.

Similarly to string annotations, pylint should just ignore these cases and let type checker handle it.

### Pylint version

```shell
pylint 2.14.0-b1
astroid 2.12.0-dev0
Python 3.9.12
```
